.github/workflows/enzyme-ci.yml

```diff
@@ -21,18 +21,35 @@ jobs:
 
     timeout-minutes: 600
     steps:
-      - name: checkout the source code
-        uses: actions/checkout@v3
+      - name: Checkout Rust source
+        uses: actions/checkout@v4
         with:
           fetch-depth: 2
-      - name: build
+      - uses: dtolnay/rust-toolchain@nightly
+      - name: Get LLVM commit hash
+        id: llvm-commit
+        run: echo "HEAD=$(git -C src/llvm-project rev-parse HEAD)" >> $GITHUB_OUTPUT
+      - name: Cache LLVM
+        id: cache-llvm
+        uses: actions/cache@v4
+        with:
+          path: build/build/x86_64-unknown-linux-gnu/llvm
+          key: ${{ matrix.os }}-llvm-${{ steps.llvm-commit.outputs.HEAD }}
+      - name: Build
         run: |
-          mkdir build
+          mkdir -p build
           cd build
+          rm -f config.toml
           ../configure --enable-llvm-link-shared --enable-llvm-plugins --enable-llvm-enzyme --release-channel=nightly --enable-llvm-assertions --enable-lld --enable-option-checking --enable-ninja --disable-docs
           ../x.py build --stage 1 library/std library/proc_macro library/test tools/rustdoc
-          rustup toolchain link enzyme `pwd`/build/`rustup target list --installed`/stage1
-          rustup toolchain install nightly # enables -Z unstable-options
-      - name: test
+          rustup toolchain link enzyme build/host/stage1
+      - name: checkout Enzyme/rustbook
+        uses: actions/checkout@v4
+        with:
+          repository: EnzymeAD/rustbook
+          ref: mdbook-test
+          path: rustbook
+      - name: test Enzyme/rustbook
+        working-directory: rustbook
         run: |
-          cargo +enzyme test --examples
+          cargo +enzyme test --workspace
```

Comment:
This is using the `mdbook-test` branch because https://github.com/EnzymeAD/rustbook/pull/3/ has not merged yet. We can switch to the default branch once it merges. Note that these tests currently fail due to the recent regression.

Reply:
Okay, this is merged (thanks @ZuseZ4) so I'll switch back to`ref:main`.

Reply:
mered